src/core_init_atmosphere/mpas_geotile_manager.F

```diff
@@ -0,0 +1,1025 @@
+        inquire(file=trim(mgr % directory)//"index", exist=res)
+        if (.not. res) then
+            call mpas_log_write("Could not find an 'index' file in geotile directory: "//trim(path), messageType=MPAS_LOG_ERR)
+            ierr = 1
+            return
+        endif
+        mgr % index = trim(mgr % directory)//"index"
+
+        ! Create the pool for this geotile and call mpas_parse_index
+        call mpas_pool_create_pool(mgr % pool)
+        ierr = mpas_parse_index(mgr % index, mgr % pool)
+        if (ierr /= 0) then
+            call mpas_log_write("Error parsing geotile index file: "//trim(mgr % index), messageType=MPAS_LOG_ERR)
+            ierr = 1
+            return
+        endif
+
+        signed => null()
+        endian => null()
+        scalefactor => null()
+
+        err_level = mpas_pool_get_error_level()
+        call mpas_pool_set_error_level(MPAS_POOL_SILENT)
+
+        call mpas_pool_get_config(mgr % pool, 'signed', signed)
+        call mpas_pool_get_config(mgr % pool, 'endian', endian)
+        call mpas_pool_get_config(mgr % pool, 'scale_factor', scalefactor)
+
+
+        !
+        ! signed, endian, and scale_factor all have default values, so if they are
+        ! not present in the index file then set them as the default values, as reported
+        ! in section 3-53 of the WRF-AWR User's Guide
+        !
+        if (.not. associated(endian)) then
+            allocate(endian)
+            endian = "big"
+            call mpas_pool_add_config(mgr % pool, 'endian', endian)
+        endif
+
+        if (.not. associated(scalefactor)) then
+            allocate(scalefactor)
+            scalefactor = 1.0_RKIND
+            call mpas_pool_add_config(mgr % pool, 'scale_factor', scalefactor)
+        endif
+
+        if (.not. associated(signed)) then
+            allocate(signed)
+            signed = 0 ! 0 = 'no', 1 = 'yes'
+            call mpas_pool_add_config(mgr % pool, 'signed', signed)
+        endif
+
+        ! Reset the pool's error level
+        call mpas_pool_set_error_level(err_level)
+
+        call mpas_pool_get_config(mgr % pool, 'tile_x', tile_nx)
+        call mpas_pool_get_config(mgr % pool, 'tile_y', tile_ny)
+        call mpas_pool_get_config(mgr % pool, 'dx', dx)
+        call mpas_pool_get_config(mgr % pool, 'dy', dy)
+
+        ! Calculate the total number of pixels in x dir
+        ! NOTE: This calculation assumes that a dataset is a global dataset and may
+        ! not work correctly for non-global datasets
+        mgr % pixel_nx = nint(360.0_RKIND / dx)
+        mgr % pixel_ny = nint(180.0_RKIND / dy)
+
+        ! Calculate the number of tiles in the x, y directions
+        ! NOTE: This calculation assumes that a dataset is a global dataset and may
+        ! not work correctly for non-global datasets
+        mgr % nTileX = int((360.0_RKIND / dx)) / tile_nx
+        mgr % nTileY = int((180.0_RKIND / dy)) / tile_ny
```

Comment:
As with the calculation of `pixel_nx` and `pixel_ny`, I think we need to use `nint` here rather than `int`. Better still, we could consider just using `mgr % pixel_nx` and `mgr % pixel_ny` rather than duplicating the computation of those variables.

Reply:
Using `mgr % pixel_nx/ny` is a great idea!